### ISSUE TITLE
dependency llvm not found for llvm-14 (missing suffix variant when searching config tool)  

### ISSUE THREAD

**Issue by Nic30** [mesonbuild/meson#10396]
**Describe the bug**
llvm-config-14 is not found by `dependency('llvm', version : '>=14.0.0')`
The problem is that this list https://github.com/mesonbuild/meson/blob/1a80e5b35ed79ac67461d54346a9299df3b9c4c3/mesonbuild/environment.py#L158 is missing "-14" (please add "-15" as well).

**To Reproduce**
`docker run -i -t ubuntu /bin/bash`
`apt update && apt install llvm-14 llvm-14-dev`

```python
project( 'proj0',  ['c', 'cpp'],  version: '0.0', # FIXME: git hash (see URL above). license: 'MIT',
  meson_version: '>= 0.62.1',  default_options: [],
)
llvm_dep = dependency('llvm', version : '>=14.0.0')
```
```
meson build/
The Meson build system
Version: 0.62.1
Source dir: /
Build dir: /build
Build type: native build
Project name: proj0
Project version: 0.0
C compiler for the host machine: cc (gcc 11.2.0 "cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0")
C linker for the host machine: cc ld.bfd 2.38
C++ compiler for the host machine: c++ (gcc 11.2.0 "c++ (Ubuntu 11.2.0-19ubuntu1) 11.2.0")
C++ linker for the host machine: c++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Did not find CMake 'cmake'
Found CMake: NO
llvm-config found: NO need ['>=14.0.0']
Run-time dependency LLVM found: NO (tried config-tool)

meson.build:12:0: ERROR: CMake binary for machine 1 not found. Giving up.
```
(installing cmake makes no difference)

**Expected behavior**
If I link the llvm-config-14 to llvm-config it works
`update-alternatives --install /usr/bin/llvm-config llvm-config /usr/bin/llvm-config-14 100`
```
llvm-config found: YES (/usr/bin/llvm-config) 14.0.0
Run-time dependency LLVM found: YES 14.0.0
```
I would expect this behavior from the beginning.
For the llvm-13 it was working just fine.
 

**system parameters**
* native build
* ubuntu:latest docker image, but happens on host machine Ubuntu 22.04 LTS as well
* Python 3.10.4
* meson 0.62.1
* ninja 1.10.1

**Comment by tristan957** [comment]
@Nic30 Thanks for identifying the issue. Could you submit a quick PR :)

**Comment by Nic30** [comment]
@dcbaker already fixed this issue in 32ebb81f1a8583f46fd8f71cecbaecd3ad66b2cd

**Comment by Iktomist** [comment]
This really hasn't been fixed.
